lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
Ya, could be (stupid VM's doing smart things).  Without looking terribly deeply here, it seems that we might be able to avoid using closure scope stuff (and avoid generating a new function for each transition created too), by elevating this to a private prototype method (and moving `checkForAbort`) and calling it from the constructor..